src/Bicep.Core.Samples/Files/InvalidResources_CRLF/main.bicep

```diff
@@ -1428,3 +1428,27 @@ resource anyTypeInExistingScopeLoop 'Microsoft.Network/dnsZones/AAAA@2018-05-01'
 resource tenantLevelResourceBlocked 'Microsoft.Management/managementGroups@2020-05-01' = {
   name: 'tenantLevelResourceBlocked'
 }
+
+// #completionTest(15,36,37) -> resourceTypes
+resource comp1 'Microsoft.Resources/'
+
+// #completionTest(15,16,17) -> resourceTypes
+resource comp2 ''
+
+// #completionTest(38) -> resourceTypes
+resource comp3 'Microsoft.Resources/t'
+
+// #completionTest(40) -> resourceTypes
+resource comp4 'Microsoft.Resources/t/v'
+
+// #completionTest(49) -> resourceTypes
+resource comp5 'Microsoft.Storage/storageAccounts'
+
+// #completionTest(50) -> storageAccountsResourceTypes
+resource comp6 'Microsoft.Storage/storageAccounts@'
+
+// #completionTest(52) -> templateSpecsResourceTypes
+resource comp7 'Microsoft.Resources/templateSpecs@20'
+
+// #completionTest(60,61) -> virtualNetworksResourceTypes
+resource comp8 'Microsoft.Network/virtualNetworks@2020-06-01'

```

Comment:
Btw there is a `CompletionTests` class where you can easily write completion tests without the need to manually check the cursor positions. Instead, you can just insert `|` to the bicep source code and call a helper method to get the cursor position.

Reply:
Yeah I started using it more and more as well.